### ISSUE TITLE
Multiple planes cause the capture full of glitches

### ISSUE THREAD

**Issue by ogdenwebb** [nowrep/obs-vkcapture#21]
For sure it related to 9ef7c689f3f96dd77bf7a691572c7151e01de849 talking about Vulkan, because with b8f9543b351f3b1e7e9c3ea698cf6a5264b3e9f1 it works fine. Not sure if GL capture is working fine, but I can test it later.

An example output taken in Deep Rock Galactic, but the issue isn't game specific.

![mpv-shot0001](https://user-images.githubusercontent.com/9018005/126550334-154ec53a-64c9-4897-86eb-26ceedc65548.jpg)

Terminal output
```
info: [linux-vkcapture] Creating texture from dmabuf 1920x1080 modifier:72057594037927935
info: [linux-vkcapture]  [0] fd:111 stride:8192 offset:0
```

**Comment by nowrep** [comment]
Can you please post logs from the game?
Do you have `VK_EXT_image_drm_format_modifier` extension? (see `vulkaninfo`).

**Comment by ogdenwebb** [comment]
Log from the game.
```
[obs-vkcapture] Init 0.8
[obs-vkcapture] Injecting VK_KHR_external_memory_fd extension
[obs-vkcapture] Injecting VK_EXT_image_drm_format_modifier extension
[obs-vkcapture] DRM format modifiers support not available
```

> Do you have VK_EXT_image_drm_format_modifier extension? (see vulkaninfo).

Looks like no, shall I?

**Comment by nowrep** [comment]
> upd: AFAIK Mesa has implemented this extension only for AMD GFX9+ and I have RX 580, which is GFX8, so probably no.

Yes, that's correct. When I added support for it, it only worked on my Intel GPU, so without much looking into it I just thought it's not supported in radv.
But now I upgraded from polaris to navi2 and the extension is there, which means it goes through the drm modifier code path and I missed that small mistake I made that only affects the "legacy" (without that extension) code path.
